docs/core/diagnostics/cli-tools/dotnet-trace.md

```diff
@@ -0,0 +1,151 @@
+                1 - Critical
+                2 - Error
+                3 - Warning
+                4 - Informational
+                5 - Verbose
+            KeyValueArgs            - A semicolon separated list of key=value
+        KeyValueArgs format: '[key1=value1][;key2=value2]'
+            note: values that contain ';' or '=' characters should be surrounded by double quotes ("), e.g., 'key="value;with=symbols";key2=value2'
+
+  --buffersize <Size>
+    Sets the size of the in-memory circular buffer in megabytes. Default 256 MB.
+
+  -f, --format
+    The format of the output trace file.  This defaults to "nettrace" on Windows and "speedscope" on other OSes.
+```
+
+## Collect a trace with `dotnet-trace`
+
+To collect traces, using `dotnet-trace`, you'll need to:
+
+- First, find out the process identifier (pid) of the .NET Core application to collect traces from.
+
+  - On Windows, there are options such as using the task manager or the `tasklist` command on the cmd window.
+  - On Linux, the trivial option could be using `pidof` on the terminal window.
+
+You may also use the command `dotnet-trace list-processes` command to find out what .NET Core processes are running, along with their process IDs.
+
+- Then, run the following command:
+
+```bash
+dotnet-trace collect --process-id <PID>
+
+Press <Enter> to exit...
+Connecting to process: <Full-Path-To-Process-Being-Profiled>/dotnet.exe
+Collecting to file: <Full-Path-To-Trace>/trace.nettrace
+  Session Id: <SessionId>
+  Recording trace 721.025 (KB)
+```
+
+- Finally, stop collection by pressing the `<Enter>` key, and `dotnet-trace` will finish logging events to *trace.nettrace* file.
+
+## Viewing the trace captured from `dotnet-trace`
+
+On Windows, `.nettrace` files can be viewed on [PerfView](https://github.com/microsoft/perfview) for analysis, just like traces collected with ETW or LTTng. For traces collected on Linux, you can either move the trace to a Windows machine to be viewed on PerfView.
+
+You may also view the trace on a Linux machine by changing the output format of `dotnet-trace` to `speedscope`. You can change the output file format using the `-f|--format` option - `-f speedscope` will make `dotnet-trace` to produce a `speedscope` file. You can currently choose between `nettrace` (the default option) and `speedscope`. `Speedscope` files can be opened at https://www.speedscope.app.
+
+Note: The .NET Core runtime generates traces in the `nettrace` format, and they're converted to speedscope (if specified) after the trace is completed. Since some conversions may result in loss of data, the original `nettrace` file is preserved next to the converted file.
+
+## Using `dotnet-trace` to collect counter values over time
+
+If you're trying to use `EventCounter` for basic health monitoring in  performance-sensitive settings like production environments and you want to collect traces instead of watching them in real time, you can do that with `dotnet-trace` as well.
```

Comment:
```suggestion
If you're trying to use `EventCounter` for basic health monitoring in performance-sensitive settings like production environments and you want to collect traces instead of watching them in real time, you can do that with `dotnet-trace` as well.
```